src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/resthandler/RestIndexPolicyAction.kt

```diff
@@ -0,0 +1,158 @@
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.IndexStateManagementPlugin.Companion.INDEX_STATE_MANAGEMENT_INDEX
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.IndexStateManagementPlugin.Companion.INDEX_STATE_MANAGEMENT_TYPE
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.IndexStateManagementPlugin.Companion.POLICY_BASE_URI
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.models.Policy
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.models.Policy.Companion.POLICY_TYPE
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.util.REFRESH
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.util._ID
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.util._VERSION
+import org.elasticsearch.action.ActionListener
+import org.elasticsearch.action.DocWriteRequest
+import org.elasticsearch.action.admin.indices.create.CreateIndexResponse
+import org.elasticsearch.action.index.IndexRequest
+import org.elasticsearch.action.index.IndexResponse
+import org.elasticsearch.action.support.WriteRequest
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.common.lucene.uid.Versions
+import org.elasticsearch.common.settings.Settings
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.rest.BaseRestHandler
+import org.elasticsearch.rest.BaseRestHandler.RestChannelConsumer
+import org.elasticsearch.rest.BytesRestResponse
+import org.elasticsearch.rest.RestChannel
+import org.elasticsearch.rest.RestController
+import org.elasticsearch.rest.RestRequest
+import org.elasticsearch.rest.RestRequest.Method.PUT
+import org.elasticsearch.rest.RestResponse
+import org.elasticsearch.rest.RestStatus
+import org.elasticsearch.rest.action.RestActions
+import org.elasticsearch.rest.action.RestResponseListener
+import java.io.IOException
+import java.time.Instant
+
+class RestIndexPolicyAction(
+    settings: Settings,
+    controller: RestController,
+    indexStateManagementIndices: IndexStateManagementIndices
+) : BaseRestHandler(settings) {
+    private var ismIndices = indexStateManagementIndices
+
+    init {
+        controller.registerHandler(PUT, POLICY_BASE_URI, this)
+        controller.registerHandler(PUT, "$POLICY_BASE_URI/{policyID}", this)
+        ismIndices = indexStateManagementIndices
+    }
+
+    override fun getName(): String {
+        return "index_policy_action"
+    }
+
+    @Throws(IOException::class)
+    override fun prepareRequest(request: RestRequest, client: NodeClient): RestChannelConsumer {
+        val id = request.param("policyID", Policy.NO_ID)
+        if (Policy.NO_ID == id) {
+            throw IllegalArgumentException("Missing policy ID")
+        }
+
+        val xcp = request.contentParser()
+        val policy = Policy.parseWithType(xcp, id).copy(lastUpdatedTime = Instant.now())
+        val policyVersion = RestActions.parseVersion(request)
```

Comment:
Use `seq_no` and `primary_term` for the updates.

Reply:
This hasn't been updated to use 6.7 yet because of the dependency on Job Scheduler.
I believe Job Scheduler recently provided an update to 7.0 so will do a separate PR another time to convert this to 7.0

Reply:
Oaky sounds good.